gateway/src/apicast/policy/rate_limit/README.md

```diff
@@ -0,0 +1,183 @@
+# Rate Limit policy
+
+- [**Description**](#description)
+- [**Types of limits**](#types-of-limits)
+- [**Limit definition**](#limit-definition)
+- [**Liquid templating**](#liquid-templating)
+- [**Requests over limits**](#requests-over-limits)
+- [**Per-gateway vs shared**](#per-gateway-vs-shared)
+- [**Complete config example**](#complete-config-example)
+
+
+## Description
+
+The purpose of this policy is to control the rate of traffic sent to the
+upstream API. There are several kinds of limits that can be defined. Here are
+some examples:
+
+- The client with address 127.0.0.1 can access the specific endpoint
+`/something` up to 100 times per minute.
+- Requests that contain a JSON web token (JWT) whose `sub` claim has the value
+`abc` can make up to 10 calls to the API per second.
+
+
+## Types of limits
+
+The types of limits supported by the policy are the ones provided by the
+[lua-resty-limit-traffic](https://github.com/openresty/lua-resty-limit-traffic)
+library:
+
+- leaky_bucket_limiters: based on the "leaky_bucket" algorithm, which is based
+on an average number of requests plus a maximum burst size.
+- fixed_window_limiters: based on a fixed window of time (last X seconds).
+- connection_limiters: based on the concurrent number of connections.
+
+Any limit can be scoped by service or globally.
+
+
+## Limit definition
+
+The limits have a key that encodes the entities that we want to use to define
+the limit (an IP, a service, an endpoint, an ID, the value for a specific
+header, etc.). Each limit also has some parameters that vary depending on the
+type:
+
+- leaky_bucket_limiters: `rate`, `burst`. Can make up to `rate` requests per
```

Comment:
It might be worth saying it will try to slow down the incoming request that was between `rate` and `burst`.

Reply:
:+1: